src/dataflow/src/source/mod.rs

```diff
@@ -1512,7 +1521,7 @@ fn maybe_emit_timestamp_bindings(
     source_name: &str,
     worker_id: usize,
     timestamp_bindings_updater: &mut Option<TimestampBindingUpdater>,
-    timestamp_histories: &mut TimestampBindingRc,
```

Comment:
Commenting on this line because I can't comment further below:

I think theres still something missing, or a little bit confusing going on here. Specifically - line 1530 (the call to update) dumps out all of the available timestamp bindings but I believe what we actually want it to do is to dump out all of the bindings for times less than the current durability frontier

Reply:
because i think as it stands, we will hold back the capability time to the durability frontier, but then send all bindings at that capability time, which i think will violate "the contents of persist collection of bindings always trail behind whats in sqlite)

Reply:
Are you saying that holding back the frontier does not communicate to the downstream operator that it should not simply persist anything it receives?

Reply:
summarizing from slack but:

The `TimestampBIndingUpdater` constructs the diff between the last time someone called `update`. and the current call to `update` and generates a set of `((source offset, assigned_ts), ts, diff)` triples where the `ts` is the current bindings capability time. Even if we hold back that `ts` to the durability frontier, it will still emit `(source_offset, assigned_ts)` pairs where `assigned_ts` is in advance of the durability frontier

Reply:
@aljoscha is the expert here, but I think one potential way to make this work

- have the binding triples `((source offset, assigned_ts), assigned_ts, diff)`
- only emit new bindings after a certain frontier (basically do the same logic we do to send bindings to coord)

Reply:
I think this should work as is. All the additional durability capabilities do is hold back the frontier for downstream operators. They don't mess with how the source operator otherwise manages its capabilities.

What this means for bindings (and data) that we emit is:
 - Yes, we emit data and bindings that are beyond the durability frontier.
 - **Yes, those will be written to persistence.**
 - Holding back the downstream frontier (via the additional durability capabilities) will make sure that we **don't seal the bindings/data that we have already written down in persistence**. And this is good because we only consider sealed data to be valid when restarting/rehydrating our state.

There is still one problem with this PR, though, we need to downgrade the bindings/data capabilities to the persistence seal frontier, otherwise we will get errors when trying to persist bindings/data at a timestamp that is not beyond the persistence seal frontier. This (and the original problem) can be re-produced with the instructions from here: https://github.com/MaterializeInc/materialize/issues/10404. I'll push a commit that fixes this last problem, I hope that's alright!